### PR TITLE
Suppress ComparisonOutOfRange errorprone warning in AbstractGasLimitSpecification

### PR DESCRIPTION
**Pull request by usmansaleem**
## PR description
Suppress ComparisonOutOfRange error-prone warning in AbstractGasLimitSpecification. This was reported during errorprone plugin update. Add a unit test to cover the max limit as well.

### Thanks for sending a pull request! Have you done the following?

- [x] Checked out our [contribution guidelines](https://github.com/hyperledger/besu/blob/main/CONTRIBUTING.md)?
- [x] Considered documentation and added the `doc-change-required` label to this PR [if updates are required](https://wiki.hyperledger.org/display/BESU/Documentation).
- [x] Considered the changelog and included an [update if required](https://wiki.hyperledger.org/display/BESU/Changelog).
- [x] For database changes (e.g. KeyValueSegmentIdentifier) considered compatibility and performed forwards and backwards compatibility tests

### Most advanced CI tests are deferred until PR approval, but you could:

- [x] locally run all unit tests via: `./gradlew build`
- [x] locally run all acceptance tests via: `./gradlew acceptanceTest`
- [x] locally run all integration tests via: `./gradlew integrationTest`
- [x] locally run all reference tests via: `./gradlew ethereum:referenceTests:referenceTests`

### CODE REVIEW

File: ethereum/core/src/main/java/org/hyperledger/besu/ethereum/mainnet/AbstractGasLimitSpecification.java [hunk 1]
```diff
@@ -40,7 +40,14 @@ public static long deltaBound(final long currentGasLimit) {
...
+  /**
+   * Verify that the target gas limit is within the allowed bounds.
+   *
+   * @param targetGasLimit the target gas limit to validate
+   * @return true if within bounds
+   */
   public static boolean isValidTargetGasLimit(final long targetGasLimit) {
-    return DEFAULT_MIN_GAS_LIMIT <= targetGasLimit && DEFAULT_MAX_GAS_LIMIT >= targetGasLimit;
+    // only check the lower bound as the upper bound is Long.MAX_VALUE
+    return targetGasLimit >= DEFAULT_MIN_GAS_LIMIT;
```
Comment (siladu) [inline ethereum/core/src/main/java/org/hyperledger/besu/ethereum/mainnet/AbstractGasLimitSpecification.java:51]: Can you point to which tests this is covered by please?

Comment (usmansaleem) [reply ethereum/core/src/main/java/org/hyperledger/besu/ethereum/mainnet/AbstractGasLimitSpecification.java:51]: ethereum/core/src/test/java/org/hyperledger/besu/ethereum/mainnet/TargetingGasLimitCalculatorTest.java

Comment (siladu) [reply ethereum/core/src/main/java/org/hyperledger/besu/ethereum/mainnet/AbstractGasLimitSpecification.java:51]: Thanks. I find this errorprone check odd - wouldn't this code break if we error changed the DEFAULT_MAX to be something other than Long.MAX_VALUE?

Comment (macfarla) [reply ethereum/core/src/main/java/org/hyperledger/besu/ethereum/mainnet/AbstractGasLimitSpecification.java:51]: agree it seems safer to suppress this warning

Comment (usmansaleem) [reply ethereum/core/src/main/java/org/hyperledger/besu/ethereum/mainnet/AbstractGasLimitSpecification.java:51]: To counter-argue, if we jump from long to BigInteger for example, then the targetGasLimit would also change from long to BigInteger. What is the value `DEFAULT_GAS_MAX_LIMIT` as per the spec? Here it seems to be bounded by the variable type we are using i.e. Long.MAX_VALUE.

I'll close this PR and will add the suppression when we bring in the updated errorprone library dependency.

Comment (siladu) [reply ethereum/core/src/main/java/org/hyperledger/besu/ethereum/mainnet/AbstractGasLimitSpecification.java:51]: Can't find a max value in the spec, was looking here: https://ethereum.github.io/execution-specs/src/ethereum/shanghai/fork.py.html#ethereum.shanghai.fork.check_gas_limit:0

It's UInt so specified as arbitrary size https://ethereum.github.io/execution-specs/src/ethereum/base_types.py.html

>  if we ever changed the DEFAULT_MAX to be something other than Long.MAX_VALUE

I was more thinking if a future spec introduced a specific max value to enforce something in the protocol.

Comment (usmansaleem) [reply ethereum/core/src/main/java/org/hyperledger/besu/ethereum/mainnet/AbstractGasLimitSpecification.java:51]: Agreed. I'll add a unit test instead to cover the max value.

**Review by macfarla — APPROVED** [review]

**Review by siladu — APPROVED** [review]
👍

**Pull request merged**